katib-ui-rock/rockcraft.yaml

```diff
@@ -19,36 +19,86 @@ platforms:
   amd64:
 
 parts:
-  katib-ui:
+  frontend-lib:
+    plugin: nil
+    source: https://github.com/kubeflow/kubeflow
+    source-type: git
+    # Set source-tag here to the commit cited in https://github.com/kubeflow/katib/blob/master/pkg/ui/v1beta1/frontend/COMMIT
+    # for this Katib release
+    # Note: Katib 0.16.0 cannot build with Kubeflow v1.8-branch because Kubeflow refactored this lib
+    source-commit: 046c6d3c8301a95a2809c0d927bbbf72638dd639
+    # Omit source-depth because we're cloning a commit
+    # source-depth: 1
+    build-environment:
+      # suppress analytics collection
+      - NG_CLI_ANALYTICS: false
+    build-snaps:
+      - node/12/stable
+    override-build: |
+      # Clean the build dir so we don't accidentally use files we didn't intend
+      rm -rf ./*
+
+      # Pull only the files we need for the build from src, and build
+      cp $CRAFT_PART_SRC/components/crud-web-apps/common/frontend/kubeflow-common-lib/package*.json ./
+      npm ci
+      cp -r $CRAFT_PART_SRC/components/crud-web-apps/common/frontend/kubeflow-common-lib/* ./
+      npm run build
+      
+    override-stage: |
+      mkdir -p $CRAFT_STAGE/frontend-lib/dist
+      cp -r $CRAFT_PART_BUILD/dist/kubeflow $CRAFT_STAGE/frontend-lib/dist/kubeflow
+
+  frontend:
+    after: [frontend-lib]
+    plugin: nil
+    source: https://github.com/kubeflow/katib
+    source-type: git
+    source-tag: v0.16.0
+    source-depth: 1
+    build-snaps:
+      - node/12/stable
+    override-build: |
+      # Clean the build dir so we don't accidentally use files we didn't intend
+      rm -rf ./*
+
+      # Pull only the files we need for the build from src
+      cp $CRAFT_PART_SRC/pkg/ui/v1beta1/frontend/package*.json ./
+      npm ci
+      cp -r $CRAFT_STAGE/frontend-lib/dist/kubeflow ./node_modules/kubeflow
+      cp -r $CRAFT_PART_SRC/pkg/ui/v1beta1/frontend/* .
+      npm run build:prod
+
+    override-prime: |
+      mkdir -p $CRAFT_PRIME/app/build
+      cp -r $CRAFT_PART_BUILD/dist/static $CRAFT_PRIME/app/build/static
+
+  backend:
     plugin: go
     source: https://github.com/kubeflow/katib
     source-type: git
-    source-tag: v0.15.0
+    source-tag: v0.16.0
     build-snaps:
-      - go
-      - curl
-    build-packages:
-      - npm
-    stage-packages:
-      - libc6_libs
+      - go/1.20/stable
     build-environment:
-      - CGO_ENABLED: "0"
-      - GOOS: "linux"
-      - GOARCH: "amd64"
```

Comment:
What about `GOARCH`?

Reply:
Hmm looking from how they build their images in CI ([here](https://github.com/kubeflow/katib/blob/v0.16.0/.github/workflows/publish-core-images.yaml)), I see they do not define this anywhere.Is that why you skipped it too?

Reply:
Yeah I omitted it just because they do, but I don't know if that is appropriate.

Reply:
I think it's fine since we build the image ourselves. I mean, even if we add it, that won't change something since in their case, I guess the CI could at some point provide that variable in order to build images for another Architecture. In our case, I guess building for a different architecture would require to modify other parts of rockcraft.yaml.

That being said, I think we should document either inside the rockcraft.yaml or the issue responsible for it some of this information, so one won't have to look at the review comments to understand why this is omitted.

Reply:
k i'll add a note to the issue